### PR TITLE
DM-40118: Produce more useful source injection data reduction logs

### PR DESCRIPTION
**Pull request by leeskelvin**

### CODE REVIEW

File: python/lsst/source/injection/inject_base.py [hunk 1]
```diff
@@ -199,7 +199,12 @@ def run(self, injection_catalogs, input_exposure, psf, photo_calib, wcs):
 
         # Make empty table if none supplied to support process_all_data_ids.
         if len(injection_catalogs) == 0:
-            injection_catalogs = [Table()]
+            if self.config.process_all_data_ids:
+                injection_catalogs = [Table(names=["ra", "dec", "source_type"])]
+            else:
+                raise RuntimeError(
+                    "No injection catalogs overlapping this DatasetRef. Check your input catalog coverage."
```
Comment (timj) [inline python/lsst/source/injection/inject_base.py:206]: The run() method does not refer to a DatasetRef does it? So I'm not convinced this error message should mention a DatasetRef.

Comment (leeskelvin) [reply python/lsst/source/injection/inject_base.py:206]: The `injection_catalogs` input comes from the `inject_visit` `runQuantum`, which is just a `butlerQC.get(input_refs)`. If this returns nothing, then there is no overlap coverage for the query you've supplied. What would be more appropriate wording here? "...overlapping your data query" ?

Comment (leeskelvin) [reply python/lsst/source/injection/inject_base.py]: Changed to:
```
No injection sources overlap the data query. Check injection catalog coverage.
```

**Comment by jtmccann** [comment]
LGTM!

**Pull request merged**